sormas-ui/src/main/java/de/symeda/sormas/ui/utils/components/CheckboxSet.java

```diff
@@ -0,0 +1,244 @@
+	protected Component initContent() {
+		layout = new VerticalLayout();
+		layout.setWidthFull();
+		layout.setMargin(new MarginInfo(true, false));
+
+		if (items != null) {
+			rows = buildCheckboxRows(items);
+		}
+
+		return layout;
+	}
+
+	public void setItems(List<T> items, Function<T, String> groupingFunction) {
+		this.items = items;
+		this.groupingFunction = groupingFunction;
+
+		if (layout != null) {
+			layout.removeAllComponents();
+			rows = buildCheckboxRows(items);
+		}
+	}
+
+	private List<CheckboxRow> buildCheckboxRows(List<T> items) {
+		List<String> groups;
+		if (groupingFunction != null) {
+			groups = items.stream().map(i -> groupingFunction.apply(i)).distinct().collect(Collectors.toList());
+		} else {
+			groups = Collections.singletonList(null);
+		}
+
+		return groups.stream().map((group) -> {
+			List<CheckboxRow> groupRows;
+			if (group != null) {
+				groupRows = createRows(items.stream().filter(i -> group.equals(groupingFunction.apply(i))).collect(Collectors.toList()));
+				layout.addComponent(createGroupHeader(group, groupRows));
+			} else {
+				groupRows = createRows(items);
+			}
+
+			layout.addComponents(groupRows.toArray(new Component[] {}));
+			return groupRows;
+		}).flatMap(List::stream).collect(Collectors.toList());
+	}
+
+	private List<CheckboxRow> createRows(List<T> groupItems) {
+		List<CheckboxRow> rows = new ArrayList<>();
+		for (int i = 0, size = groupItems.size(); i < size; i += 2) {
+			T item1 = groupItems.get(i);
+			T item2 = i < size - 1 ? groupItems.get(i + 1) : null;
+
+			CheckboxRow checkboxRow = new CheckboxRow(item1, item2);
+
+			checkboxRow.checkBoxes.forEach(cb -> cb.addValueChangeListener(e -> {
+				fireEvent(new CheckboxValueChangeEvent(cb));
+			}));
+
+			rows.add(checkboxRow);
+		}
+
+		return rows;
+	}
+
+	private HorizontalLayout createGroupHeader(String group, List<CheckboxRow> rows) {
+		Label groupLabel = new Label(group);
+		groupLabel.addStyleName(CssStyles.H3);
+
+		Label tickForAllLAbel = new Label(I18nProperties.getString(Strings.checkboxSetTickAnAnswerForAll));
+		tickForAllLAbel.addStyleName(CssStyles.LABEL_ITALIC);
+
+		Button yesAllBurron = ButtonHelper.createButton(
+			I18nProperties.getCaption(Captions.actionYesAll),
+			(e) -> rows.forEach(r -> r.setValueForAll(true)),
+			ValoTheme.BUTTON_LINK,
+			CssStyles.BUTTON_COMPACT);
+		Button noAllButton = ButtonHelper.createButton(
+			I18nProperties.getCaption(Captions.actionNoAll),
+			(e) -> rows.forEach(r -> r.setValueForAll(false)),
+			ValoTheme.BUTTON_LINK,
+			CssStyles.BUTTON_COMPACT);
+
+		HorizontalLayout buttonsLayout = new HorizontalLayout(tickForAllLAbel, yesAllBurron, noAllButton);
+		buttonsLayout.setWidthFull();
+		buttonsLayout.setExpandRatio(tickForAllLAbel, 1);
+
+		HorizontalLayout headerLayout = new HorizontalLayout(groupLabel, buttonsLayout);
+		headerLayout.setWidthFull();
+
+		headerLayout.setExpandRatio(groupLabel, 0.5f);
+		headerLayout.setExpandRatio(buttonsLayout, 0.5f);
+
```

Comment:
Add `headerLayout.setComponentAlignment(buttonsLayout, Alignment.MIDDLE_RIGHT);` to have the "all" buttons aligned with the caption.